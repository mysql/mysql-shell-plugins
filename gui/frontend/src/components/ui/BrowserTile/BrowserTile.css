/*
 * Copyright (c) 2020, 2022, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

.msg.browserTile {
    display: flex;
    flex-direction: row;
    position: relative;

    align-items: stretch;

    margin: 10px;
    padding: 0 0 0 20px;
    width: 270px;
    height: 80px;

    border-radius: 4px;
    border: 2px solid var(--browserTile-border);

    color: var(--browserTile-foreground);
    background-color: var(--browserTile-background);

    transition: background .15s ease-in;
    user-select: none;
}

/* Need a stronger focus indicator for this big buttons. A single thin line doesn't work well. */
.msg.browserTile:focus {
    outline: none;
}

.msg.browserTile:focus::after {
    content: '';
    display: block;
    position: absolute;
    top: -6px;
    bottom: -6px;
    left: -6px;
    right: -6px;
    border-radius: 6px;
    border: 2px dashed var(--focusBorder);
    overflow: visible;
}

.msg.browserTile .icon:not(.codicon) {
    flex: 0 0 40px;
    margin-right: 16px;

    background-color: var(--browserTile-foreground);
}

.msg.browserTile:not(:disabled).dropTarget,
.msg.browserTile:not(:disabled):hover {
    background-color: var(--browserTile-hoverBackground);
}

.msg.browserTile:not(:disabled):active,
.msg.browserTile:focus,
.msg.browserTile:focus-within {
    color: var(--browserTile-activeForeground);
    border-color: var(--browserTile-activeBorder);
    background-color: var(--browserTile-activeBackground);
}

.msg .browserTile.secondary {
    border-color: var(--browserTileSecondary-border);
    background-color: var(--browserTileSecondary-background);
    color: var(--browserTileSecondary-foreground);
}

.msg.browserTile.secondary:not(:disabled):active,
.msg.browserTile.secondary:focus {
    color: var(--browserTileSecondary-activeForeground);
    border-color: var(--browserTileSecondary-activeBorder);
    background-color: var(--browserTileSecondary-activeBackground);
}

.msg.browserTile .textHost {
    flex: 1 1 auto;
    justify-content: center;
    padding-right: 8px;
}

.msg.browserTile .label {
    margin: 0;
    padding: 0;

    text-align: left;

    color: var(--browserTile--foreground);
}

.msg.browserTile .tileCaption {
    font-size: 1rem;
    font-weight: 600;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.msg.browserTile .tileDescription {
    font-size: 0.7rem;
}

.msg.browserTile.secondary .label {
    color: var(--browserTileSecondary--foreground);
}

.msg.browserTile.secondary .icon {
    background-color: var(--browserTileSecondary-foreground);
}

.msg.browserTile #actions {
    display: flex;
    visibility: hidden;
    position: absolute;
    justify-items: center;

    top: 0;
    right: 0;
    margin: 0;

    width: 24px;
    height: 24px;

    border: none;
    border-radius: 0 0 0 4px;
    background: var(--browserTile-border);
}

.msg.browserTile:focus #actions,
.msg.browserTile:hover #actions {
    visibility: visible;
}

.msg.browserTile #actions>#triggerEditMenu {
    flex: 1 1 auto;

    min-width: 0;
    width: 17px;
    height: 17px;

    padding: 0;
    margin: 0;

    color: var(--icon-foreground);
    font-weight: 800;

    text-align: center;
    border: none;
    background: var(--browserTile-border);
}
