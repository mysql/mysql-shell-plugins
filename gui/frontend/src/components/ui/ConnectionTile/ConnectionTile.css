/*
 * Copyright (c) 2025, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

.msg.connectionTile {
    display: flex;
    flex-direction: row;
    position: relative;

    align-items: stretch;

    margin: 2px;
    padding: 0 0 0 20px;
    width: 270px;
    height: 80px;

    border-radius: 4px;
    border: 2px solid var(--connectionTile-border);

    color: var(--connectionTile-foreground);
    background-color: var(--connectionTile-background);

    transition: background .15s ease-in;
    user-select: none;
}

/* Need a stronger focus indicator for this big buttons. A single thin line doesn't work well. */
.msg.connectionTile:focus,
.msg.connectionTile:focus-within {
    outline: none;
    border-color: var(--connectionTileCreateNew-activeBackground);
    color: var(--connectionTile-activeForeground);
    background-color: var(--connectionTile-activeBackground);
}

.msg.connectionTile:focus::after {
    content: '';
    display: block;
    position: absolute;
    top: -4px;
    bottom: -4px;
    left: -4px;
    right: -4px;
    border-radius: 4px;
    border: 2px dashed var(--focusBorder);
    overflow: visible;
}

.msg.button.connectionTile .icon {
    flex: 0 0 40px;
    margin-right: 16px;

    /* for codicons */
    font-size: 28pt;
    align-content: center;

    background-color: var(--connectionTile-foreground);
}

.msg.connectionTile:not(:disabled).dropTarget,
.msg.connectionTile:not(:disabled):hover {
    background-color: var(--connectionTile-hoverBackground);
}

.msg.connectionTile:not(:disabled):active {
    color: var(--connectionTile-activeForeground);
    border-color: var(--connectionTile-activeBorder);
    background-color: var(--connectionTile-activeBackground);
}

.msg.connectionTile.createNew:not(:disabled).dropTarget,
.msg.connectionTile.createNew:not(:disabled):hover {
    background-color: var(--connectionTileCreateNew-hoverBackground);
}

.msg.connectionTile.createNew:not(:disabled):active {
    color: var(--connectionTileCreateNew-activeForeground);
    border-color: var(--connectionTileCreateNew-activeBorder);
    background-color: var(--connectionTileCreateNew-activeBackground);
}

.msg.connectionTile.createNew {
    border-color: var(--connectionTileCreateNew-border);
    background-color: var(--connectionTileCreateNew-background);
    color: var(--connectionTileCreateNew-foreground);
}

.msg.connectionTile.createNew:focus {
    color: var(--connectionTileCreateNew-activeForeground);
    border-color: transparent;
    background-color: var(--connectionTileCreateNew-activeBackground);
}

.msg.connectionTile.group {
    border-color: var(--connectionTileGroup-border);
    background-color: var(--connectionTileGroup-background);
    color: var(--connectionTileGroup-foreground);
}

.msg.connectionTile.group:not(:disabled).dropTarget,
.msg.connectionTile.group:not(:disabled):hover {
    background-color: var(--connectionTileGroup-hoverBackground);
}

.msg.connectionTile.group:not(:disabled):active {
    color: var(--connectionTileGroup-activeForeground);
    border-color: var(--connectionTileGroup-activeBorder);
    background-color: var(--connectionTileGroup-activeBackground);
}

.msg.connectionTile.group:focus {
    color: var(--connectionTileGroup-activeForeground);
    border-color: transparent;
    background-color: var(--connectionTileGroup-activeBackground);
}

.msg.connectionTile.back {
    border-color: var(--connectionTileBack-border);
    background-color: var(--connectionTileBack-background);
    color: var(--connectionTileBack-foreground);
}

.msg.connectionTile.back:not(:disabled).dropTarget,
.msg.connectionTile.back:not(:disabled):hover {
    background-color: var(--connectionTileBack-hoverBackground);
}

.msg.connectionTile.back:not(:disabled):active {
    color: var(--connectionTileBack-activeForeground);
    border-color: var(--connectionTileBack-activeBorder);
    background-color: var(--connectionTileBack-activeBackground);
}

.msg.connectionTile.back:focus {
    color: var(--connectionTileBack-activeForeground);
    border-color: transparent;
    background-color: var(--connectionTileBack-activeBackground);
}

.msg.connectionTile .textHost {
    flex: 1 1 auto;
    justify-content: center;
    padding-right: 39px;
}

.msg.connectionTile .label {
    margin: 0;
    padding: 0;

    text-align: left;

    color: var(--connectionTile-foreground);
}

.msg.connectionTile .tileCaption {
    font-size: 1.15rem;
    font-weight: 600;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.msg.connectionTile .tileDescription {
    font-size: 0.8rem;
}

.msg.connectionTile.secondary .label {
    color: var(--connectionTileCreateNew--foreground);
}

.msg.connectionTile.secondary .icon {
    background-color: var(--connectionTileCreateNew-foreground);
}

.msg.connectionTile #actionsBackground {
    visibility: hidden;
    position: absolute;

    top: 0;
    right: 0;
    margin: 0;

    width: 39px;
    height: 76px;

    border: none;
    background: var(--connectionTile-border);
    opacity: 35%;
}

.msg.connectionTile #actions {
    display: flex;
    visibility: hidden;
    position: absolute;
    justify-items: center;

    top: 0;
    right: 0;
    margin: 0;

    width: 39px;
    height: 76px;

    border: none;
    background: none;
}

.msg.connectionTile:focus #actions,
.msg.connectionTile:hover #actions,
.msg.connectionTile:focus #actionsBackground,
.msg.connectionTile:hover #actionsBackground {
    visibility: visible;
}

.msg.connectionTile #actions > #triggerEditMenu {
    flex: 1 1 auto;

    min-width: 0;
    width: 17px;
    height: 17px;

    padding: 0;
    margin: 0;

    color: var(--icon-foreground);
    font-weight: 800;

    text-align: center;
    border: none;
    background: var(--connectionTile-border);
}

.msg.connectionTile #actions .button {
    margin: 0;
    padding: 0;
    min-width: 0;
    background: none;
    border: none;
    color: var(--connectionTile-foreground);
    width: 24px;
    height: 24px;

    opacity: 100%;
}

.msg.connectionTile #actions .icon {
    flex: 0 0 16px;
    height: 16px;
    margin: 0;
}
