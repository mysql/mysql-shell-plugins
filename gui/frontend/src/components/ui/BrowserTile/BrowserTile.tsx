/*
 * Copyright (c) 2020, 2024, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import "./BrowserTile.css";

import { ComponentChild, createRef } from "preact";

import { Container, ContentAlignment, Orientation } from "../Container/Container.js";
import { filterInt } from "../../../utilities/string-helpers.js";
import { IComponentProperties, ComponentBase, DragEventType } from "../Component/ComponentBase.js";
import { Icon } from "../Icon/Icon.js";
import { Label } from "../Label/Label.js";
import { Button } from "../Button/Button.js";
import { KeyboardKeys } from "../../../utilities/helpers.js";

export enum BrowserTileType {
    Open,
    CreateNew,
}

export interface ITileActionOptions {
    newTab?: boolean;
    target?: HTMLElement | null;
    editor?: "notebook" | "script";

    id?: string;
    endpoints?: Array<{ ipAddress: string; }>;
    displayName?: string;
    description?: string;
    compartmentId?: string;
    profileName?: string;
}

export interface IBrowserTileProperties extends IComponentProperties {
    innerRef?: preact.RefObject<HTMLDivElement>;

    tileId: number;
    caption: string;
    description: string;
    type: BrowserTileType;
    icon: string;

    onAction?: (action: string, props: IBrowserTileProperties, options: ITileActionOptions) => void;
    onTileReorder?: (draggedTileId: number, props: unknown) => void;
}

export abstract class BrowserTile<P extends IBrowserTileProperties> extends ComponentBase<P> {

    private actionsRef = createRef<HTMLDivElement>();

    protected abstract renderTileActionUI: () => ComponentChild;

    public constructor(props: P) {
        super(props);

        this.addHandledProperties("innerRef", "tileId", "caption", "description", "type", "icon", "onAction",
            "onTileReorder");
        this.connectDragEvents();
    }

    public render(): ComponentChild {
        const { innerRef, tileId, type, icon, caption, description } = this.props;

        const className = this.getEffectiveClassNames([
            "browserTile",
            this.classFromProperty(type === BrowserTileType.CreateNew, "secondary"),
        ]);

        return (
            <Button
                innerRef={innerRef}
                id={tileId.toString()}
                className={className}
                onClick={this.handleClick}
                onKeyPress={this.handleKeydown}
                {...this.unhandledProperties}
            >
                <Icon src={icon} />
                <Container
                    orientation={Orientation.TopDown}
                    className="textHost"
                >
                    <Label className="tileCaption">{caption}</Label>
                    <Label className="tileDescription">{description}</Label>
                </Container>
                {type === BrowserTileType.Open &&
                    <>
                        <Container id="actionsBackground" />
                        <Container
                            id="actions"
                            orientation={Orientation.TopDown}
                            mainAlignment={ContentAlignment.Center}
                            crossAlignment={ContentAlignment.Center}
                            innerRef={this.actionsRef}
                        >
                            {this.renderTileActionUI()}
                        </Container>
                    </>
                }
            </Button>
        );
    }

    protected override handleDragEvent = (type: DragEventType, e: DragEvent): boolean => {
        if (!e.dataTransfer) {
            return false;
        }

        const element = e.currentTarget as HTMLElement;
        switch (type) {
            case DragEventType.Start: {
                e.dataTransfer.effectAllowed = "move";
                e.dataTransfer.setData("browser/tile", element.id);

                return true;
            }

            case DragEventType.Enter: {
                e.stopPropagation();
                e.preventDefault();

                if (element.contains(e.relatedTarget as Node)) {
                    element.classList.add("dropTarget");
                }

                return true;
            }

            case DragEventType.Leave: {
                e.stopPropagation();
                e.preventDefault();

                if (!element.contains(e.relatedTarget as Node)) {
                    element.classList.remove("dropTarget");
                }

                return true;
            }

            case DragEventType.Drop: {
                element.classList.remove("dropTarget");

                const id = filterInt(e.dataTransfer.getData("browser/tile"));
                if (!isNaN(id)) {
                    const { onTileReorder } = this.props;
                    onTileReorder?.(id, this.props);
                }

                return true;
            }

            default: {
                return false;
            }
        }

    };

    private handleClick = (e: MouseEvent | KeyboardEvent): void => {
        const { onAction, type } = this.props;

        const event = e as MouseEvent;
        const button = event.currentTarget as HTMLButtonElement;

        // Have to prevent double clicks on browser tiles. But since everything is async there's no way to know
        // the action triggered by the tile is finished (or at least started, so the button is hidden).
        // Hence the only way to enable the button is to use a timer.
        button.disabled = true;
        setTimeout(() => {
            button.disabled = false;
        }, 200);
        e.stopPropagation();

        if (type === BrowserTileType.Open) {
            onAction?.("open", this.props, { newTab: event.metaKey || event.altKey });
        } else {
            onAction?.("new", this.props, {});
        }
    };

    private handleKeydown = (e: KeyboardEvent): void => {
        if (e.key === KeyboardKeys.A) { // Unmodified A key.
            e.stopPropagation();
            const { type, onAction } = this.props;
            if (type === BrowserTileType.Open) {
                this.actionsRef.current?.focus();
                onAction?.("menu", this.props, { target: this.actionsRef.current });
            }
        }
    };
}
