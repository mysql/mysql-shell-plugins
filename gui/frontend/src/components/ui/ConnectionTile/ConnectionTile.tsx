/*
 * Copyright (c) 2020, 2023, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is also distributed with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have included with MySQL.
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import { ComponentChild } from "preact";

import toolbarMenuIcon from "../../../assets/images/toolbar/toolbar-menu.svg";
import newNotebookIcon from "../../../assets/images/newNotebook.svg";
import newScriptIcon from "../../../assets/images/newScript.svg";

import { IConnectionDetails } from "../../../supplement/ShellInterface/index.js";
import { BrowserTile, IBrowserTileProperties } from "../BrowserTile/BrowserTile.js";
import { Button } from "../Button/Button.js";
import { Icon } from "../Icon/Icon.js";

export interface IConnectionTileProperties extends IBrowserTileProperties {
    details: IConnectionDetails;
}

export class ConnectionTile extends BrowserTile<IConnectionTileProperties> {

    public static defaultProps = {
        className: "connectionTile",
    };

    public constructor(props: IConnectionTileProperties) {
        super(props);

        this.addHandledProperties("details");
    }

    public render(): ComponentChild {
        return super.render();
    }

    protected renderTileActionUI = (): ComponentChild => {
        return (
            <>
                <Button
                    id="tileMoreActionsAction"
                    data-tooltip="More Actions"
                    imageOnly
                    onClick={this.handleActionClick}
                >
                    <Icon src={toolbarMenuIcon} data-tooltip="inherit" />
                </Button>
                <Button
                    id="tileNewNotebookAction"
                    data-tooltip="Create New Notebook"
                    imageOnly
                    onClick={this.handleActionClick}
                >
                    <Icon src={newNotebookIcon} data-tooltip="inherit" />
                </Button>
                <Button
                    id="tileNewScriptAction"
                    data-tooltip="Create New Script"
                    imageOnly
                    onClick={this.handleActionClick}
                >
                    <Icon src={newScriptIcon} data-tooltip="inherit" />
                </Button>
            </>
        );
    };

    private handleActionClick = (e: MouseEvent | KeyboardEvent): void => {
        const { onAction } = this.mergedProps;

        e.stopPropagation();
        const id = (e.currentTarget as HTMLElement).id;
        switch (id) {
            case "tileMoreActionsAction": {
                onAction?.("menu", this.mergedProps, { target: e.currentTarget as HTMLElement });

                break;
            }

            case "tileNewNotebookAction": {
                onAction?.("open", this.mergedProps, { newTab: e.metaKey || e.altKey, editor: "notebook" });

                break;
            }

            case "tileNewScriptAction": {
                onAction?.("open", this.mergedProps, { newTab: e.metaKey || e.altKey, editor: "script" });

                break;
            }

            default:
        }
    };

}
