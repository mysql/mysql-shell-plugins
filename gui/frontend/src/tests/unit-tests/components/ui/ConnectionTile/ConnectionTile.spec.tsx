/*
 * Copyright (c) 2022, 2024, Oracle and/or its affiliates.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2.0,
 * as published by the Free Software Foundation.
 *
 * This program is designed to work with certain software (including
 * but not limited to OpenSSL) that is licensed under separate terms, as
 * designated in a particular file or component or in included license
 * documentation.  The authors of MySQL hereby grant you an additional
 * permission to link the program and your derivative works with the
 * separately licensed software that they have either included with
 * the program or referenced in the documentation.
 *
 * This program is distributed in the hope that it will be useful,  but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
 * the GNU General Public License, version 2.0, for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 */

import mysqlIcon from "../../../../../assets/images/file-icons/scriptMysql.svg";

import { mount } from "enzyme";

import { DBType, IConnectionDetails } from "../../../../../supplement/ShellInterface/index.js";
import { BrowserTileType } from "../../../../../components/ui/BrowserTile/BrowserTile.js";
import { ConnectionTile } from "../../../../../components/ui/ConnectionTile/ConnectionTile.js";

describe("ConnectionTile component tests", (): void => {

    const details: IConnectionDetails = {
        id: 1,
        dbType: DBType.MySQL,
        caption: "Tile 1",
        description: "A description",
        options: {},
    };

    it("Test ConnectionTile output (Snapshot)", () => {
        const component = mount<ConnectionTile>(
            <ConnectionTile
                id="tile1"
                details={details}
                tileId={1}
                caption={details.caption}
                description={details.description}
                type={BrowserTileType.Open}
                icon={mysqlIcon}

            />,
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });
});
