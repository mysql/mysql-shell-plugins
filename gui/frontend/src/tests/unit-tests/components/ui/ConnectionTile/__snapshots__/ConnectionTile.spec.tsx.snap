// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTile component tests Test ConnectionTile output (Snapshot) 1`] = `
<ConnectionTile
  caption="Tile 1"
  className="connectionTile"
  description="A description"
  details={
    {
      "caption": "Tile 1",
      "dbType": "MySQL",
      "description": "A description",
      "id": 1,
      "options": {},
    }
  }
  icon="scriptMysql.svg"
  id="tile1"
  tileId={1}
  type={0}
>
  <Button
    className="msg browserTile connectionTile"
    id="tile1"
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onKeyPress={[Function]}
  >
    <div
      className="msg button browserTile connectionTile"
      draggable={true}
      id="tile1"
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onKeyPress={[Function]}
      onMouseDown={[Function]}
      role="button"
      style={
        {
          "flexDirection": undefined,
        }
      }
      tabIndex={0}
    >
      <Icon
        disabled={false}
        src="scriptMysql.svg"
      >
        <div
          className="msg icon"
          style={
            {
              "WebkitMaskImage": "url(scriptMysql.svg)",
              "backgroundColor": undefined,
              "height": undefined,
              "maskImage": "url(scriptMysql.svg)",
              "width": undefined,
            }
          }
        />
      </Icon>
      <Container
        className="textHost"
        orientation="column"
      >
        <div
          className="msg container fixedScrollbar textHost"
          style={
            {
              "alignItems": undefined,
              "flexDirection": "column",
              "flexWrap": undefined,
              "justifyContent": undefined,
            }
          }
        >
          <Label
            className="tileCaption"
          >
            <label
              className="msg label tileCaption"
              data-tooltip="expand"
              style={{}}
            >
              Tile 1
            </label>
          </Label>
          <Label
            className="tileDescription"
          >
            <label
              className="msg label tileDescription"
              data-tooltip="expand"
              style={{}}
            >
              A description
            </label>
          </Label>
        </div>
      </Container>
      <Container
        id="actionsBackground"
        orientation="row"
      >
        <div
          className="msg container fixedScrollbar"
          id="actionsBackground"
          style={
            {
              "alignItems": undefined,
              "flexDirection": "row",
              "flexWrap": undefined,
              "justifyContent": undefined,
            }
          }
        />
      </Container>
      <Container
        crossAlignment="center"
        id="actions"
        innerRef={
          {
            "current": <div
              class="msg container fixedScrollbar"
              id="actions"
              style="flex-direction: column; justify-content: center; align-items: center;"
            >
              <div
                class="msg button imageOnly"
                data-tooltip="More Actions"
                draggable="true"
                id="tileMoreActionsAction"
                role="button"
                tabindex="0"
              >
                <div
                  class="msg icon"
                  data-tooltip="inherit"
                  style="mask-image: url(toolbar-menu.svg);"
                />
              </div>
              <div
                class="msg button imageOnly"
                data-tooltip="Create New Notebook"
                draggable="true"
                id="tileNewNotebookAction"
                role="button"
                tabindex="0"
              >
                <div
                  class="msg icon"
                  data-tooltip="inherit"
                  style="mask-image: url(newNotebook.svg);"
                />
              </div>
              <div
                class="msg button imageOnly"
                data-tooltip="Create New Script"
                draggable="true"
                id="tileNewScriptAction"
                role="button"
                tabindex="0"
              >
                <div
                  class="msg icon"
                  data-tooltip="inherit"
                  style="mask-image: url(newScript.svg);"
                />
              </div>
            </div>,
          }
        }
        mainAlignment="center"
        orientation="column"
      >
        <div
          className="msg container fixedScrollbar"
          id="actions"
          style={
            {
              "alignItems": "center",
              "flexDirection": "column",
              "flexWrap": undefined,
              "justifyContent": "center",
            }
          }
        >
          <Button
            data-tooltip="More Actions"
            id="tileMoreActionsAction"
            imageOnly={true}
            onClick={[Function]}
          >
            <div
              className="msg button imageOnly"
              data-tooltip="More Actions"
              draggable={true}
              id="tileMoreActionsAction"
              onClick={[Function]}
              onMouseDown={[Function]}
              role="button"
              style={
                {
                  "flexDirection": undefined,
                }
              }
              tabIndex={0}
            >
              <Icon
                data-tooltip="inherit"
                disabled={false}
                src="toolbar-menu.svg"
              >
                <div
                  className="msg icon"
                  data-tooltip="inherit"
                  style={
                    {
                      "WebkitMaskImage": "url(toolbar-menu.svg)",
                      "backgroundColor": undefined,
                      "height": undefined,
                      "maskImage": "url(toolbar-menu.svg)",
                      "width": undefined,
                    }
                  }
                />
              </Icon>
            </div>
          </Button>
          <Button
            data-tooltip="Create New Notebook"
            id="tileNewNotebookAction"
            imageOnly={true}
            onClick={[Function]}
          >
            <div
              className="msg button imageOnly"
              data-tooltip="Create New Notebook"
              draggable={true}
              id="tileNewNotebookAction"
              onClick={[Function]}
              onMouseDown={[Function]}
              role="button"
              style={
                {
                  "flexDirection": undefined,
                }
              }
              tabIndex={0}
            >
              <Icon
                data-tooltip="inherit"
                disabled={false}
                src="newNotebook.svg"
              >
                <div
                  className="msg icon"
                  data-tooltip="inherit"
                  style={
                    {
                      "WebkitMaskImage": "url(newNotebook.svg)",
                      "backgroundColor": undefined,
                      "height": undefined,
                      "maskImage": "url(newNotebook.svg)",
                      "width": undefined,
                    }
                  }
                />
              </Icon>
            </div>
          </Button>
          <Button
            data-tooltip="Create New Script"
            id="tileNewScriptAction"
            imageOnly={true}
            onClick={[Function]}
          >
            <div
              className="msg button imageOnly"
              data-tooltip="Create New Script"
              draggable={true}
              id="tileNewScriptAction"
              onClick={[Function]}
              onMouseDown={[Function]}
              role="button"
              style={
                {
                  "flexDirection": undefined,
                }
              }
              tabIndex={0}
            >
              <Icon
                data-tooltip="inherit"
                disabled={false}
                src="newScript.svg"
              >
                <div
                  className="msg icon"
                  data-tooltip="inherit"
                  style={
                    {
                      "WebkitMaskImage": "url(newScript.svg)",
                      "backgroundColor": undefined,
                      "height": undefined,
                      "maskImage": "url(newScript.svg)",
                      "width": undefined,
                    }
                  }
                />
              </Icon>
            </div>
          </Button>
        </div>
      </Container>
    </div>
  </Button>
</ConnectionTile>
`;
